src/deriveNativeScriptHash.h

```diff
@@ -0,0 +1,52 @@
+#ifndef H_CARDANO_APP_DERIVE_NATIVE_SCRIPT_HASH
+#define H_CARDANO_APP_DERIVE_NATIVE_SCRIPT_HASH
+
+#include "bip44.h"
+#include "cardano.h"
+#include "common.h"
+#include "handlers.h"
+#include "nativeScriptHashBuilder.h"
+
+handler_fn_t deriveNativeScriptHash_handleAPDU;
+
+// a special type for distinguishing what to show in the UI, makes it easier
+// to handle PUBKEY DEVICE_OWNED vs THIRD_PARTY
+typedef enum {
+	UI_SCRIPT_PUBKEY_PATH = 0,
+	UI_SCRIPT_PUBKEY_HASH,
+	UI_SCRIPT_ALL,
+	UI_SCRIPT_ANY,
+	UI_SCRIPT_N_OF_K,
+	UI_SCRIPT_INVALID_BEFORE,
+	UI_SCRIPT_INVALID_HEREAFTER,
+} ui_native_script_type;
```

Comment:
I'm not sure we really need this. Wouldn't a single if for ui_script_pubkey disctinction solve it?

Reply:
It also makes it easier to get the header, because for that I use the array called `ui_native_script_header` which translates this enum to a header for each of the script types, so if it's possible I would keep this, because it simplifies life for me in two separate places.